Add scene platform for Sunricher DALI integration

niracler opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Add scene platform for Sunricher DALI integration

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request:  https://github.com/home-assistant/home-assistant.io/pull/42096
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.
- [x] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.  https://github.com/maginawin/PySrDaliGateway/compare/v0.16.2...v0.18.0

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

niracler committed:
feat(sunricher_dali): add scene platform

- Implement DaliCenterScene entity with activation support
- Add parallel discovery of devices and scenes using asyncio.gather
- Include entity ID mappings in scene attributes for related lights
- Add comprehensive test coverage with fixtures and snapshots

niracler committed:
feat(sunricher_dali): add scene platform

- Implement DaliCenterScene entity with activation support
- Add parallel discovery of devices and scenes using asyncio.gather
- Include entity ID mappings in scene attributes for related lights
- Add comprehensive test coverage with fixtures and snapshots

niracler committed:
add PARALLEL_UPDATES

niracler committed:
fix(sunricher_dali): remove debug logging and improve test coverage

- Remove debug warning log in scene.py
- Fix test_scene_read_failure to properly test exception handling during setup
- Achieve 100% code coverage for scene platform

niracler committed:
refactor(sunricher_dali): optimize test structure

- Unify test data with DEVICE_DATA constant
- Simplify mock device creation
- Merge redundant test cases (9 → 5)
- Maintain 100% test coverage

joostlek commented on homeassistant/components/sunricher_dali/scene.py:
```diff
@@ -0,0 +1,116 @@
+            "area_id": scene_details["area_id"],
+            "channel": scene_details["channel"],
+            "entity_id": [],
+        }
```
What does the user need this for?
serial number is already in the device
scene id?
area_id would also be in the device?
What's channel for?

niracler committed:
Add base entity for Sunricher DALI

joostlek commented on homeassistant/components/sunricher_dali/entity.py:
```diff
@@ -0,0 +1,53 @@
+        if isinstance(dali_object, Device):
+            self._attr_available = dali_object.status == "online"
+        else:
+            self._attr_available = True
```
You could consider just creating a second base entity that inherits `DaliCenterEntity` and call it `DaliCenterDeviceEntity`. This way you dont have to do an isinstance check

niracler committed:
deps: update PySrDaliGateway to 0.18.0

niracler committed:
refactor(sunricher_dali): extract DaliDeviceEntity class

- Remove isinstance check from DaliCenterEntity
- Add DaliDeviceEntity for device-specific initialization
- Update DaliCenterLight to use new base class

niracler committed:
Merge branch 'dev' into feat/sunricher-dali-scene

niracler committed:
Merge remote-tracking branch 'upstream/dev' into feat/sunricher-dali-scene

joostlek committed:
Fix

joostlek reviewed (approved):
I removed the entity id mapping for now so we can get this merged, feel free to open a new PR and we can discuss that change in that PR
